### PR TITLE
[INLONG-8228][DataProxy] Optimize the implementation of the index output to files

### LINKED ISSUE #8228: [Improve][DataProxy] Optimize the implementation of the index output to files
### Description

At present, there are several problems in the implementation of index  output:
1. The MonitorIndex and MonitorIndexExt used in the index output file consume a lot of time in index calculation and need to be optimized;
2. Statistical indexs do not distinguish between the output of summary information and detailed information;
3. The key of the error index is not detailed enough, and some output needs to be refined

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by gosonzhang**
- Fixes #8228

1. Add MonitorIndex and MonitorStats classes, adjust statistics and output methods;
2. Add detailed indicator processing logic and API;
3. Refine the key of the error indicator, and distinguish different error scenarios according to the indicator key
4. Add SO_BACKLOG, SO_REUSEADDR, SO_LINGER parameter settings for netty;
5. Other optimization and modification

### CODE REVIEW

**Review by baomingyu — APPROVED** [review]
+1

**Review by dockerzhang — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorStats.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+package org.apache.inlong.dataproxy.metrics.stats;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.LongAdder;
+
+public class MonitorStats extends AbsStatsDaemon {
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorStats.java:27]: Please add javadocs for the classes file, thx.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorIndex.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.LongAdder;
+
+public class MonitorIndex extends AbsStatsDaemon {
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorIndex.java:28]: ditto.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorIndex.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    private static final Logger logger = LoggerFactory.getLogger(MonitorIndex.class);
+
+    private static final AtomicLong recId = new AtomicLong(0);
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorIndex.java:32]: It is recommended to define static final constants as UPPERCASE_LETTERS.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorIndex.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    public MonitorIndex(String name, long intervalMill, int maxCnt) {
+        super(name, intervalMill, maxCnt);
+        this.statsUnits[0] = new StatsUnit(name);
+        this.statsUnits[1] = new StatsUnit(name);
+    }
+
+    /**
+     * addSuccStats
+     *
+     * @param key
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorIndex.java:44]: As we can see, if there is no explanation after the param parameter in the javadoc, there will be a warning message in some IDEs; in addition, the javadoc without description has no practical meaning.

So, please add description information for these parameters, or delete the declaration of these params in javadoc.

image

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorStats.java (hunk 1 above)
Comment (gosonzhang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorStats.java:33]: done

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorIndex.java (hunk 2 above)
Comment (gosonzhang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/stats/MonitorIndex.java:34]: done

**Comment by gosonzhang** [comment]
@healchow, please review again

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**